src/Illuminate/Collections/Collection.php

```diff
@@ -402,7 +405,7 @@ public function forget($keys)
      *
      * @param  mixed  $key
      * @param  mixed  $default
-     * @return mixed
+     * @return T
```

Comment:
this is wrong because it can also return the type of default

Reply:
its also wrong to just make default T... a common thing people do is default to a different type... like null :)

Reply:
Hold on, I'll propose an implementation for this.

Reply:
We can do something like https://github.com/laravel/framework/pull/38042/commits/0d324cede268abeb89d4c1b49b312ad798f9fe9a
, would that be acceptable?